src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -825,6 +825,54 @@ - (void)invalidateCASESessionForNode:(chip::NodeId)nodeID;
 }
 @end
 
+/**
+ * Shim to allow us to treat an MTRDevicePairingDelegate as an
+ * MTRDeviceControllerDelegate.
+ */
+@interface MTRDevicePairingDelegateShim : NSObject <MTRDeviceControllerDelegate>
+@property (nonatomic, readonly) id<MTRDevicePairingDelegate> delegate;
+- (instancetype)initWithDelegate:(id<MTRDevicePairingDelegate>)delegate;
+@end
+
+@implementation MTRDevicePairingDelegateShim
+- (instancetype)initWithDelegate:(id<MTRDevicePairingDelegate>)delegate
+{
+    if (self = [super init]) {
+        _delegate = delegate;
+    }
+    return self;
+}
+
+- (void)onStatusUpdate:(MTRCommissioningStatus)status
```

Comment:
The shim should probably override `respondsToSelector:` to call `self.delegate respondsToSelector:` for the selectors defined in `MTRDeviceControllerDelegate`, so we don't end up dispatching blocks for methods that aren't implemented. You could then remove the `respondsToSelector:` checks from the actual methods that do the forwarding.

Reply:
> The shim should probably override `respondsToSelector:` to call `self.delegate respondsToSelector:` for the selectors defined in `MTRDeviceControllerDelegate`

Ah, and `super respondsToSelector` otherwise?  Yes, can do that.